agent/consul/leader_connect.go

```diff
@@ -259,6 +261,48 @@ func retryLoopBackoffHandleSuccess(ctx context.Context, loopFn func() error, err
 	}
 }
 
+func retryLoopBackoffPeering(ctx context.Context, loopFn func() error, errFn func(error), abortOnSuccess bool) {
+	var failedAttempts uint
+	var err error
+	limiter := rate.NewLimiter(loopRateLimit, retryBucketSize)
+	for {
+		// Rate limit how often we run the loop
+		limiter.Wait(ctx)
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+		if (1 << failedAttempts) < maxRetryBackoff {
+			failedAttempts++
+		}
+		retryTime := (1 << failedAttempts) * time.Second
+
+		if err = loopFn(); err != nil {
+			errFn(err)
+
+			if err == peerstream.ConnectedWithFollowerErr {
+				fmt.Printf("setting retry to 10ms\n")
+				retryTime = 10 * time.Millisecond
+			}
```

Comment:
I made an earlier comment below about moving some of the error-checking here so we don't pass around an extra sentinel error. I think in an ideal case, any gRPC errors with `FailedPrecondition` should be retried quickly but I haven't combed through all the possible errors to see if that's true or not.

Reply:
Implemented that!